test/verify-conditions.spec.js

```diff
@@ -29,118 +33,245 @@ describe("verify conditions", () => {
         getRepositoryBranch.mockResolvedValue(undefined);
     });
 
-    it("should verify that Helm is installed by checking the version", async () => {
-        await verifyConditions(pluginConfig, context);
+    describe("aws checks", () => {
+        let expectedBucketName,
+            expectedCallerIdentity;
 
-        expect(helmVersion).toHaveBeenCalled();
-    });
+        beforeEach(() => {
+            expectedBucketName = chance.word();
+            expectedCallerIdentity = chance.word();
+            expectedPluginConfig = createAWSPluginConfig(expectedBucketName);
 
-    it("should verify that each chart passes `helm lint`", async () => {
-        await verifyConditions(pluginConfig, context);
+            getCallerIdentity.mockResolvedValue(expectedCallerIdentity);
+            s3HeadBucket.mockResolvedValue(undefined);
+        });
 
-        pluginConfig.charts.forEach((chart) => {
-            expect(helmLint).toHaveBeenCalledWith(chart);
+        it("should verify AWS authentication by getting the caller identity", async () => {
+            await verifyConditions(expectedPluginConfig, context);
+
+            expect(getCallerIdentity).toHaveBeenCalledWith(expectedPluginConfig.aws.region);
         });
-    });
 
-    it("should verify that GitHub pages is enabled for the repository", async () => {
-        await verifyConditions(pluginConfig, context);
+        it("should verify access to the specified s3 bucket", async () => {
+            await verifyConditions(expectedPluginConfig, context);
 
-        expect(getRepository).toHaveBeenCalledWith(expectedRepoOwner, expectedRepoName);
-    });
+            expect(s3HeadBucket).toHaveBeenCalledWith(expectedPluginConfig.aws.region, expectedPluginConfig.aws.bucket);
+        });
 
-    it("should verify that the specified branch exists", async () => {
-        await verifyConditions(pluginConfig, context);
+        describe("when the AWS region is not specified", () => {
+            beforeEach(() => {
+                delete expectedPluginConfig.aws.region;
+            });
 
-        expect(getRepositoryBranch).toHaveBeenCalledWith(expectedRepoOwner, expectedRepoName, pluginConfig.githubPagesBranch);
-    });
+            it("should throw an error", async () => {
+                await expect(() => verifyConditions(expectedPluginConfig, context)).rejects.toThrow(AggregateError);
+                await expect(() => verifyConditions(expectedPluginConfig, context)).rejects.toThrow("Expected `aws.region`, `aws.bucket`, and `aws.bucketUrl` config options to be set");
```

Comment:
Would it be worth it to define the the exception messages in a module and de-duplicate the strings? It always make me uncomfortable when tests depend on strings that might get changed to fix grammar or improve readability it logging.

Reply:
I agree, but I think I'll follow up with this later so I can get this merged and released.